pandas/core/categorical.py

```diff
@@ -2107,6 +2108,38 @@ def _get_codes_for_values(values, categories):
     return coerce_indexer_dtype(t.lookup(vals), cats)
 
 
+def _recode_for_categories(codes, old_categories, new_categories):
+    """
+    Convert a set of codes for to a new set of categories
+
+    Parameters
+    ----------
+    codes : array
+    old_categories, new_categories : Index
+
+    Returns
+    -------
+    new_codes : array
+
+    Examples
+    --------
+    >>> old_cat = pd.Index(['b', 'a', 'c'])
```

Comment:
isn't this just a special case of ``union_categoricals``?

Reply:
these both remap codes; seems like they've should share

Reply:
I see. There was a little bit I could extract from union_categorical. Simplified things a bit too by using take_1d. See my latest commit.

Reply:
looks nice.